diff --git a/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp b/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
index 616c8a17dd6..f2bce756f47 100644
--- a/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
+++ b/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp
@@ -566,13 +566,13 @@ extern "C" LLVMRustResult LLVMRustOptimize(
   }
 
   std::optional<PGOOptions> PGOOpt;
-#if LLVM_VERSION_LT(22, 0)
+#if LLVM_VERSION_LT(21, 0)
   auto FS = vfs::getRealFileSystem();
 #endif
   if (PGOGenPath) {
     assert(!PGOUsePath && !PGOSampleUsePath);
     PGOOpt = PGOOptions(
-#if LLVM_VERSION_GE(22, 0)
+#if LLVM_VERSION_GE(21, 0)
         PGOGenPath, "", "", "", PGOOptions::IRInstr, PGOOptions::NoCSAction,
 #else
         PGOGenPath, "", "", "", FS, PGOOptions::IRInstr, PGOOptions::NoCSAction,
@@ -581,7 +581,7 @@ extern "C" LLVMRustResult LLVMRustOptimize(
   } else if (PGOUsePath) {
     assert(!PGOSampleUsePath);
     PGOOpt = PGOOptions(
-#if LLVM_VERSION_GE(22, 0)
+#if LLVM_VERSION_GE(21, 0)
         PGOUsePath, "", "", "", PGOOptions::IRUse, PGOOptions::NoCSAction,
 #else
         PGOUsePath, "", "", "", FS, PGOOptions::IRUse, PGOOptions::NoCSAction,
@@ -589,7 +589,7 @@ extern "C" LLVMRustResult LLVMRustOptimize(
         PGOOptions::ColdFuncOpt::Default, DebugInfoForProfiling);
   } else if (PGOSampleUsePath) {
     PGOOpt =
-#if LLVM_VERSION_GE(22, 0)
+#if LLVM_VERSION_GE(21, 0)
         PGOOptions(PGOSampleUsePath, "", "", "", PGOOptions::SampleUse,
 #else
         PGOOptions(PGOSampleUsePath, "", "", "", FS, PGOOptions::SampleUse,
@@ -598,7 +598,7 @@ extern "C" LLVMRustResult LLVMRustOptimize(
                    DebugInfoForProfiling);
   } else if (DebugInfoForProfiling) {
     PGOOpt = PGOOptions(
-#if LLVM_VERSION_GE(22, 0)
+#if LLVM_VERSION_GE(21, 0)
         "", "", "", "", PGOOptions::NoAction, PGOOptions::NoCSAction,
 #else
         "", "", "", "", FS, PGOOptions::NoAction, PGOOptions::NoCSAction,
@@ -1213,8 +1213,8 @@ LLVMRustCreateThinLTOData(LLVMRustThinLTOModule *modules, size_t num_modules,
   // being lifted from `lib/LTO/LTO.cpp` as well
   DenseMap<GlobalValue::GUID, const GlobalValueSummary *> PrevailingCopy;
   for (auto &I : Ret->Index) {
-#if LLVM_VERSION_GE(22, 0)
-    const auto &SummaryList = I.second.getSummaryList();
+#if LLVM_VERSION_GE(21, 0)
+    const auto &SummaryList = I.second.SummaryList;
 #else
     const auto &SummaryList = I.second.SummaryList;
 #endif
@@ -1250,8 +1250,8 @@ LLVMRustCreateThinLTOData(LLVMRustThinLTOModule *modules, size_t num_modules,
   // linkage will stay as external, and internal will stay as internal.
   std::set<GlobalValue::GUID> ExportedGUIDs;
   for (auto &List : Ret->Index) {
-#if LLVM_VERSION_GE(22, 0)
-    const auto &SummaryList = List.second.getSummaryList();
+#if LLVM_VERSION_GE(21, 0)
+    const auto &SummaryList = List.second.SummaryList;
 #else
     const auto &SummaryList = List.second.SummaryList;
 #endif
